integration_tests/tests/consistency/consistency_transaction_details_amounts.sql

```diff
@@ -0,0 +1,41 @@
+{{ config(
+    tags="fivetran_validations",
+    enabled=var('fivetran_validation_tests_enabled', false)
+) }}
+
+with prod as (
+
+    select 
+        transaction_details_id,
+        converted_amount as prod_converted_amount,
+        transaction_amount as prod_transaction_amount
+    from {{ target.schema }}_netsuite_prod.netsuite2__transaction_details
+),
+
+dev as (
+
+    select 
+        transaction_details_id,
+        converted_amount as dev_converted_amount,
+        transaction_amount as dev_transaction_amount
+    from {{ target.schema }}_netsuite_dev.netsuite2__transaction_details
+
+),
+
+final as (
+
+    select
+        prod.transaction_details_id,
+        prod.prod_converted_amount,
+        dev.dev_converted_amount,
+        prod.prod_transaction_amount,
+        dev.dev_transaction_amount
+    from prod
+    full outer join dev
+        on dev.transaction_details_id = prod.transaction_details_id
+)
```

Comment:
I know this test is passing. However, there are scenarios where if a transaction_id is not present in dev then we will just see a `null` populated. Then a valid number - `null` just returns `null` and essentially passes the where condition, but in reality there is a mismatch in the data. 

![image](https://github.com/user-attachments/assets/68e44529-3f91-40f8-badc-404538e6af8d)

See below for a suggestion to sum by the transaction date so we can look at the aggregate sum of the converted and transaction amounts. This way if there is a `null` in dev but not in prod then we should see less in the aggregate and therefore a resulting failure in the validation test. Additionally, I included a row count in here as well for extra validation. When testing locally I could see that the row counts matched between dev and prod confirming there is no fan out resulting from the new joins.

```suggestion
with prod as (

    select 
        cast(transaction_date as date) as date_day,
        count(*) as prod_row_count,
        sum(converted_amount) as prod_converted_amount,
        sum(transaction_amount) as prod_transaction_amount
    from {{ target.schema }}_netsuite_prod.netsuite2__transaction_details
    group by 1
),

dev as (

    select 
        cast(transaction_date as date) as date_day,
        count(*) as dev_row_count,
        sum(converted_amount) as dev_converted_amount,
        sum(transaction_amount) as dev_transaction_amount
    from {{ target.schema }}_netsuite_dev.netsuite2__transaction_details
    group by 1

),

final as (

    select
        prod.date_day,
        prod.prod_row_count,
        dev.dev_row_count,
        prod.prod_converted_amount,
        dev.dev_converted_amount,
        prod.prod_transaction_amount,
        dev.dev_transaction_amount
    from prod
    full outer join dev
        on dev.date_day = prod.date_day
)
```

Reply:
Updated. Made some slight changes (the most recent day can see late arriving transactions when I did a devprod run and lead to some slight discrepancies on transactions) so I created an additional filter here to get the consistency test to pass.